src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Unit.Tests/JsonSourceGeneratorIncrementalTests.cs

```diff
@@ -344,5 +347,74 @@ public static IEnumerable<object[]> GetCompilationHelperFactories()
                 .Where(m => m.ReturnType == typeof(Compilation) && m.GetParameters().Length == 0)
                 .Select(m => new object[] { Delegate.CreateDelegate(typeof(Func<Compilation>), m) });
         }
+
+        /// <summary>
+        /// Asserts for structural equality, returning a path to the mismatching data when not equal.
+        /// </summary>
+        private static void AssertStructurallyEqual<T>(T expected, T actual)
```

Comment:
Why's this generic? The same type arguments are being used AFAICT.

Reply:
It's not a bound to a particular type. The parameters could have just been `object`, however I wanted a way to ensure that both arguments are of the same type.

Reply:
https://github.com/dotnet/runtime/pull/86334#discussion_r1196803863